controller/proxy-injector/webhook.go

```diff
@@ -131,52 +131,51 @@ func Inject(linkerdNamespace string) webhook.Handler {
 			}, nil
 		}
 
+		// Resource could not be injected with the sidecar, format the reason
+		// for injection being skipped to emit an event
+		var readableReasons string
+		for _, reason := range reasons {
+			readableReasons = readableReasons + ", " + inject.Reasons[reason]
+		}
+		// removing the initial comma, space
+		readableReasons = readableReasons[2:]
+		if parent != nil {
+			recorder.Eventf(*parent, v1.EventTypeNormal, eventTypeSkipped, "Linkerd sidecar proxy injection skipped: %s", readableReasons)
+		}
+
 		// Create a patch which adds the opaque ports annotation if the workload
-		// does already have it set.
+		// doesn't already have it set.
 		patchJSON, err := resourceConfig.CreateOpaquePortsPatch()
 		if err != nil {
 			return nil, err
 		}
 
-		// If patchJSON holds a patch after checking the workload annotations,
-		// then we admit the request.
+		admissionResp := &admissionv1beta1.AdmissionResponse{
+			UID:     request.UID,
+			Allowed: true,
+		}
 		if len(patchJSON) != 0 {
+			// If resource needs to be patched with annotations (e.g opaque
+			// ports), then admit the request with the relevant patch
 			log.Infof("annotation patch generated for: %s", report.ResName())
 			log.Debugf("annotation patch: %s", patchJSON)
 			proxyInjectionAdmissionResponses.With(admissionResponseLabels(ownerKind, request.Namespace, "false", "", report.InjectAnnotationAt, configLabels)).Inc()
 			patchType := admissionv1beta1.PatchTypeJSONPatch
-			return &admissionv1beta1.AdmissionResponse{
+			admissionResp = &admissionv1beta1.AdmissionResponse{
```

Comment:
I think using early returns instead of an `else if` block actually makes the flow clearer and is more idiomatic.

Reply:
Yeah, I agree. It just seemed a bit redundant to me to return the same thing twice at the end. I've reverted back to how things were 👍🏻